### DISCUSSION TITLE
Prevent @mouseenter on page load

### DISCUSSION THREAD

**Discussion by marispro** [alpinejs/alpine#1279] [1. Help]
Hello!

Looking for help: how can I prevent @mouseenter action on page load if mouse is already on the element?
So I'm looking for solution how to make trigger mouse action only when cursor came from outside.

Any ideas? :)

**Comment by KevinBatdorf** [comment]
Maybe you could keep track of the page load and only start monitoring the event after the load event fires. OR if still no good maybe see if you can work it out using the intersection observer API

**Comment by lovedder1995** [comment]
You can try this hack

```html
<div x-data="{ fromOutside: false }">
  <div
    @mouseover="fromOutside = true"
    style="background-color: blue; padding: 10px;">
    <div
      @mouseover="if (fromOutside) alert('Mouse enter')"
      style="background-color: red; width: 500px; height: 500px;">
    </div>
  </div>
</div>
```
